src/test/java/net/logstash/logback/appender/LogstashTcpSocketAppenderTest.java

```diff
@@ -176,32 +176,51 @@ public void testReconnectOnOpen() throws Exception {
         
         appender.stop();
         
-        verify(listener, timeout(VERIFICATION_TIMEOUT)).connectionClosed(appender, socket);
-
+        verify(listener, async()).connectionClosed(appender, socket);
+        
         assertThat(appender.getConnectedDestination()).isNotPresent();
     }
 
+    
+    /**
+     * Scenario:
+     *   Failure to write in the connection output stream
+     * 
+     * Assert the appender closes the connection and reconnects
+     */
     @Test
-    public void testReconnectOnWrite() {
+    public void testReconnectOnWrite() throws Exception {
         appender.addDestination("localhost:10000");
-        appender.setReconnectionDelay(new Duration(100));
+        appender.setReconnectionDelay(Duration.buildByMilliseconds(100));
         
-        appender.start();
+        // Throw exception at first attempt to write into the connection - this should
+        // trigger a reconnect
+        doThrow(IOException.class)
+            .doNothing()
+            .when(outputStream).write(any(byte[].class), anyInt(), anyInt());
         
+        appender.start();
         verify(encoder).start();
         
-        doThrow(new RuntimeException()).doReturn("event1".getBytes(StandardCharsets.UTF_8)).when(encoder).encode(event1);
-        
         appender.append(event1);
```

Comment:
It's unclear why this test needed to change for the refactoring of the executor service.  Can you clarify?

Reply:
same reasons as above